perma_web/perma/views/user_management.py

```diff
@@ -747,26 +752,69 @@ def get_context_data(self, **kwargs):
     def form_valid(self, form):
         """ If form is submitted successfully, show success message and send email to target user. """
         response = super(BaseAddUserToGroup, self).form_valid(form)
-        if self.is_new:
-            email_new_user(
-                self.request,
-                self.object,
-                self.user_added_email_template,
-                { 'form': form }
-            )
-            messages.add_message(self.request, messages.SUCCESS,
-                                 f'<h4>Account created!</h4> <strong>{self.object.email}</strong> will receive an email with instructions on how to activate the account and create a password.',
-                                 extra_tags='safe')
+
+        def add_message(level, title, body):
+            messages.add_message(self.request, level, f'<h4>{title}</h4>{body}', extra_tags='safe')
+
+        def send_emails(users, email_function, email_template, extra_context, *args):
+            for obj in users.values():
+                try:
+                    if email_function == 'email_new_user':
+                        email_new_user(*args, obj, email_template, extra_context)
+                    else:
+                        send_user_email(obj.raw_email, email_template, extra_context)
+                except Exception as e:
```

Comment:
Thanks for adding this error handling!

As mentioned, I am a little worried that, when we are actually calling out to the mailgun API and actually sending emails, that it will be too slow to do this inline, resulting in users who upload large CSVs seeing a 502 despite the request succeeding (like happens with large link batches). I think we might need to make this async. 

Possibly to be discussed!! Even if we do, that definitely doesn't have to be part of this PR!!

Reply:
I will test in stage to see at what point we start seeing failures. Depending on that, I can add a limit to the rows.